opaque_keys/__init__.py

```diff
@@ -362,11 +392,13 @@ def __getstate__(self):
     @property
     def _key(self):
         """Returns a tuple of key fields"""
+        return unicode(self)
         # pylint: disable=no-member
-        return tuple(getattr(self, field) for field in self.KEY_FIELDS) + (self.CANONICAL_NAMESPACE, self.deprecated)
+        # return tuple(getattr(self, field) for field in self.KEY_FIELDS) + (self.CANONICAL_NAMESPACE, self.deprecated)
 
     def __eq__(self, other):
-        return isinstance(other, OpaqueKey) and self._key == other._key  # pylint: disable=protected-access
+        return unicode(self) == unicode(other)
```

Comment:
I was a little worried about correctness and collisions, given that `hash()` is going to spit out a 32-bit integer. That might be overly paranoid, considering how tightly opaque keys are constrained in terms of format.

Come to think on it, I'm still a little worried about the correctness now, since with the current code I have, unicode strings and the opaque keys they parse into will evaluate as equal, and that's some JavaScript-like madness right there. :-P

Also, I've just made an object pool for these things, so maybe we can make it even faster by just doing an `is` comparison at the top. I'll play with a couple different options and post the results.